api/client/client.go

```diff
@@ -823,15 +823,36 @@ func (c *Client) UpsertKubeService(ctx context.Context, s types.Server) error {
 // GetKubeServices returns the list of kubernetes services registered in the
 // cluster.
 func (c *Client) GetKubeServices(ctx context.Context) ([]types.Server, error) {
-	resp, err := c.grpc.GetKubeServices(ctx, &proto.GetKubeServicesRequest{}, c.callOpts...)
+	resources, err := c.GetResources(ctx, defaults.Namespace, types.KindKubeService)
 	if err != nil {
+		// DELETE IN 10.0
+		if trace.IsNotImplemented(err) {
+			resp, err := c.grpc.GetKubeServices(ctx, &proto.GetKubeServicesRequest{}, c.callOpts...)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+
+			servers := make([]types.Server, len(resp.GetServers()))
+			for i, server := range resp.GetServers() {
+				servers[i] = server
+			}
+
+			return servers, nil
+		}
```

Comment:
nit: Can you do what you did for ListNodes and move the fallback to a separate method?
```suggestion
		if trace.IsNotImplemented(err) {
			return c.getKubeServicesFallback(ctx)
		}
```